realm/realm-library/src/main/java/io/realm/internal/async/RealmThreadPoolExecutor.java

```diff
@@ -82,6 +93,27 @@ protected void beforeExecute(Thread t, Runnable r) {
         }
     }
 
+    @Override
+    protected void afterExecute(Runnable r, Throwable t) {
+        super.afterExecute(r, t);
+        tidyTransactions();
+    }
+
+    public boolean hasPendingTransactions() {
```

Comment:
Please add JavaDoc to public methods, even though it the internal API it should still be well documented.